### ISSUE TITLE
[unbound-method] conflicts with no-extra-bind

### ISSUE THREAD

**Issue by andreineculau** [typescript-eslint/typescript-eslint#1255]
**Repro**

```JSON
{
  "rules": {
    "no-extra-bind": "error",
    "@typescript-eslint/unbound-method": "error"
  }
}
```

```TS
let obj = {};
obj.fn = function() {}
// vs
// obj.fn = (function() {}).bind(obj);
```

**Expected Result**

No error.

**Actual Result**

'unbound-method' complains. But going with the unnecessary `.bind(obj)`, would turn the built-in 'no-extra-bind' rule red because there is no need 

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.7.0` |
| `@typescript-eslint/parser`        | `2.7.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.6.0` |
| `node`                             | `12.12.0` |
| `npm`                              | `6.11.3` |

**Comment by bradzacher** [comment]
I'm unable to repro your example against master.
the rule shouldn't trigger on this code, as it's an assignment, which is not an unbound usage.

**Comment by andreineculau** [comment]
I made the same mistake as in https://github.com/typescript-eslint/typescript-eslint/issues/1256 and ended up with a simplistic repro that is not good. Problem is I refactored the code where I bumped into this, and I cannot remember. Embarrassing. Sorry for the wasted time. I'll close this and open another issue when I can repro.
